src/main/webapp/app/exercises/text/participate/text-submission.service.ts

```diff
@@ -40,19 +40,30 @@ export class TextSubmissionService {
             .pipe(map((res: HttpResponse<TextSubmission>) => res.body!));
     }
 
-    getTextSubmissionsForExercise(exerciseId: number, req: { submittedOnly?: boolean; assessedByTutor?: boolean }): Observable<HttpResponse<TextSubmission[]>> {
+    getTextSubmissionsForExerciseByCorrectionRound(
+        exerciseId: number,
+        req: { submittedOnly?: boolean; assessedByTutor?: boolean },
+        correctionRound?: number,
+    ): Observable<HttpResponse<TextSubmission[]>> {
+        correctionRound = correctionRound ?? 0;
         const options = createRequestOption(req);
         return this.http
-            .get<TextSubmission[]>(`api/exercises/${exerciseId}/text-submissions`, {
+            .get<TextSubmission[]>(`api/exercises/${exerciseId}/round/${correctionRound}/text-submissions`, {
                 params: options,
                 observe: 'response',
             })
             .pipe(map((res: HttpResponse<TextSubmission[]>) => TextSubmissionService.convertArrayResponse(res)));
     }
 
-    // option = 'head': Do not optimize assessment order. Only used to check if assessments available.
-    getTextSubmissionForExerciseWithoutAssessment(exerciseId: number, option?: 'lock' | 'head'): Observable<TextSubmission> {
-        let url = `api/exercises/${exerciseId}/text-submission-without-assessment`;
+    /**
+     *
+     * @param exerciseId id of the exerciser
+     * @param option 'head': Do not optimize assessment order. Only used to check if assessments available.
+     * @param correctionRound: The correction round for which we want to get a new assessment
+     */
+    getTextSubmissionForExerciseForCorrectionRoundWithoutAssessment(exerciseId: number, option?: 'lock' | 'head', correctionRound?: number): Observable<TextSubmission> {
+        correctionRound = correctionRound ?? 0;
```

Comment:
```suggestion
    getTextSubmissionForExerciseForCorrectionRoundWithoutAssessment(exerciseId: number, option?: 'lock' | 'head', correctionRound = 0): Observable<TextSubmission> {
```

Reply:
this would default the parameter to 0, but `??` is short for: `correctionround =  correctionround : correctionround ? 0`

Reply:
Even shorter than my suggestion ^^